projects/vdk-core/src/vdk/internal/builtin_plugins/run/job_input.py

```diff
@@ -60,8 +61,8 @@ def __init__(
 
     # Connections
 
-    def get_managed_connection(self) -> ManagedConnectionBase:
-        return self.__managed_connection_builder.open_default_connection()
+    def get_managed_connection(self) -> PEP249Connection:
+        return self.__managed_connection_builder.open_default_connection().connect()
```

Comment:
Isn't `open_default_connection()` evaluating https://github.com/vmware/versatile-data-kit/blob/5f147364ab8e23a8283964b0500ba526268daff3/projects/vdk-core/src/vdk/internal/builtin_plugins/connection/impl/router.py#L71 that handles the connect?

```
if conn._is_connected():
  return conn
else:
  conn._connect()
  return conn
```

Reply:
Look at line 86,87 
When we create connection for a first time , we go through those lines. 

But that's a good point. open_connection() works differently depending if it's called once (for same db type) or twice. 
The first time it creates a connection and the second it calls conn._connect() . Which is strange. I feel it's a mistake. I certainly did not intend this on purpose.

Reply:
The change doesn't make sense. If a connection is not opened, the place that should be looked into for fixing it, is `open_connection()` method that is intended to do so.

Reply:
Yeah , I agree.